hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/metrics/ContractMetrics.java

```diff
@@ -74,77 +126,164 @@ public class ContractMetrics {
-                        final var config = new Counter.Config(METRIC_CATEGORY, name)
-                                .withDescription(descr)
-                                .withUnit(METRIC_TXN_UNIT);
-                        final var metric = newCounter(metrics, config);
-                        rejectedTxsLackingIntrinsicGas.put(txKind, metric);
-                    }
+    /**
+     * Secondary metrics are based on the system contract methods themselves, split into various categories that
+     * are based on their attributes.
+     */
+    public void createContractSecondaryMetrics() {
 
-                    // Rejected transactions for ethereum calls that are in type 3 blob transaction format
-                    {
-                        final var name = toRejectedName(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
-                        final var descr = toRejectedDescr(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
-                        final var config = new Counter.Config(METRIC_CATEGORY, name)
-                                .withDescription(descr)
-                                .withUnit(METRIC_TXN_UNIT);
-                        final var metric = newCounter(metrics, config);
-                        rejectedEthType3Counter = metric;
-                    }
-                }
-                primaryMetricsAreCreated = true;
+        if (systemContractMethodRegistry.size() == 0) {
+            // Something went wrong with the order in which components were initialized
+            log.warn("no system contract methods registered when trying to create secondary metrics");
+        }
+
+        final var contractsConfig = requireNonNull(contractsConfigSupplier.get());
+        this.p2MetricsEnabled = contractsConfig.metricsSmartContractSecondaryEnabled();
+
+        if (p2MetricsEnabled) {
+            final var metrics = requireNonNull(metricsSupplier.get());
+
+            // P2 metrics come in pairs:  a total count of something, and the error count for that same thing
+
+            // By system contract
+            // Collect all system contracts in use and create counters
+            final var allSystemContracts = systemContractMethodRegistry.allMethods().stream()
+                    .map(m -> m.systemContract().orElseThrow())
+                    .collect(Collectors.toSet());
+            for (final var systemContract : allSystemContracts) {
+                systemContractMethodCounters.put(systemContract, makeCounterPair(systemContract.name(), ""));
+            }
+
+            // By via: DIRECT vs PROXY
+            for (final var callVia : SystemContractMethod.CallVia.values()) {
+                systemContractMethodCountersVia.put(callVia, makeCounterPair(callVia.name(), ""));
+            }
+
+            // By ERC type
+
+            for (final var method : systemContractMethodRegistry.allMethods()) {
+                final var ercMembership = intersect(method.categories(), ERC_TYPES);
```

Comment:
will this not create an an entry in the map even if the intersect is empty?  I did not see a unit test to see if this would be the case.